extensions/parquet/table/src/main/java/io/deephaven/parquet/table/pagestore/ColumnChunkPageStore.java

```diff
@@ -171,17 +172,52 @@ public boolean usesDictionaryOnEveryPage() {
     @Override
     public void close() {}
 
-    FillContext innerFillContext(@NotNull final FillContext context) {
-        // TODO(deephaven-core#4836): Call this method from the appropriate place in the implementation of
-        // getPageContaining to populate the context object
-        return ((PagingContextHolder) context)
-                .updateInnerContext(this::fillContextUpdater);
+    /**
+     * Take an object of {@link PagingContextHolder} and populate the inner context with values from
+     * {@link #columnChunkReader}
+     */
+    final void innerFillContext(@Nullable final FillContext context) {
+        if (context != null) {
+            ((PagingContextHolder) context).updateInnerContext(this::fillContextUpdater);
+        }
+    }
+
+    /**
+     * Return true if this class allocated the current inner context and therefore the context can be reused for reading
+     * pages.
+     */
+    private static boolean isFillContextCompatible(@Nullable final Context currentInnerContext) {
+        return currentInnerContext instanceof ChannelContextWrapper;
+    }
+
+    static SeekableChannelsProvider.ChannelContext getChannelContext(@Nullable final FillContext fillContext) {
+        if (fillContext instanceof PagingContextHolder) {
+            final Context innerContext = ((PagingContextHolder) fillContext).getInnerContext();
+            if (innerContext instanceof ChannelContextWrapper) {
+                return ((ChannelContextWrapper) innerContext).getChannelContext();
+            }
+        }
+        return SeekableChannelsProvider.ChannelContext.NULL;
     }
 
-    private boolean isFillContextCompatible(@Nullable final Context currentInnerContext) {
-        // TODO(deephaven-core#4836): Replace this with a test to see if the fill context comes from
-        // this.ColumnChunkReader
-        return currentInnerContext == DEFAULT_FILL_INSTANCE;
+    /**
+     * Wrapper class which takes a {@link SeekableChannelsProvider.ChannelContext} and implements {@link FillContext}
+     */
+    private static class ChannelContextWrapper implements FillContext {
```

Comment:
`extends PagingContextHolder` instead of `implements FillContext` would better meet the "contract" in the `PagingChunkSource` JavaDoc. Maybe not important, since there isn't further nesting.

Reply:
I am a bit confused about this comment. Basically in the current state, `SeekableChannelsProvider.ChannelContext` does not extend from `Context`, therefore, I could not directly plug it inside a PagingContextHolder.
So I made this wrapper which can hold the ChannelContext object.

If I do `ChannelContextWrapper extends PagingContextHolder`, then I would need a `Context` object to pass in the constructor...

Reply:
```
    /**
     * Wrapper class which takes a {@link SeekableChannelsProvider.ChannelContext} and implements {@link FillContext},
     * used to update the inner context of {@link PagingContextHolder}.
     */
    private static class ChannelContextWrapper extends PagingContextHolder {
        @NotNull
        private final SeekableChannelsProvider.ChannelContext channelContext;

        public ChannelContextWrapper(
                final int chunkCapacity,
                @Nullable final SharedContext sharedContext,
                @NotNull final SeekableChannelsProvider.ChannelContext context) {
            super(chunkCapacity, sharedContext);
            this.channelContext = context;
        }

        @NotNull
        SeekableChannelsProvider.ChannelContext getChannelContext() {
            return channelContext;
        }

        @Override
        public void close() {
            channelContext.close();
        }
    }
```

```
        return (T) new ChannelContextWrapper(chunkCapacity, sharedContext, channelsProvider.makeContext());
```